{"cells":[{"cell_type":"code","execution_count":1,"metadata":{"executionInfo":{"elapsed":7,"status":"ok","timestamp":1692114479741,"user":{"displayName":"69. Ashish Yadav","userId":"08547316013865081816"},"user_tz":-330},"id":"wgcFKXXm4XMk"},"outputs":[],"source":["import pandas as pd"]},{"cell_type":"code","execution_count":2,"metadata":{"id":"RPxXM2by4gs3"},"outputs":[],"source":["df_fake = pd.read_csv(\"Fake.csv\")"]},{"cell_type":"code","execution_count":3,"metadata":{},"outputs":[],"source":["df_true = pd.read_csv(\"True.csv\")"]},{"cell_type":"code","execution_count":4,"metadata":{"id":"et4NbC2D4l5t"},"outputs":[{"name":"stdout","output_type":"stream","text":["0: (23481, 4)\n","1: (21417, 4)\n"]}],"source":["print(f\"0: {df_fake.shape}\")\n","print(f\"1: {df_true.shape}\")"]},{"cell_type":"code","execution_count":5,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["Index(['title', 'text', 'subject', 'date'], dtype='object')\n"]}],"source":["print(df_fake.columns)"]},{"cell_type":"code","execution_count":6,"metadata":{},"outputs":[],"source":["df_fake = df_fake.drop(['title','subject', 'date'],axis=1)\n","df_true = df_true.drop(['title','subject', 'date'],axis=1)"]},{"cell_type":"code","execution_count":7,"metadata":{},"outputs":[],"source":["df_fake['label'] = 0\n","df_true['label'] = 1"]},{"cell_type":"code","execution_count":8,"metadata":{},"outputs":[],"source":["df = pd.concat([df_fake,df_true])"]},{"cell_type":"code","execution_count":9,"metadata":{},"outputs":[{"data":{"text/plain":["label\n","0    23481\n","1    21417\n","Name: count, dtype: int64"]},"execution_count":9,"metadata":{},"output_type":"execute_result"}],"source":["df['label'].value_counts()"]},{"cell_type":"code","execution_count":10,"metadata":{"id":"-vpegB6T4m7I"},"outputs":[],"source":["# df.label.value_counts()"]},{"cell_type":"code","execution_count":11,"metadata":{"id":"FNo5qDfF4qGf"},"outputs":[],"source":["# df['label_num'] = df['label'].map({'Fake':0,'Real':1})\n","# df.head()"]},{"cell_type":"code","execution_count":12,"metadata":{},"outputs":[],"source":["import spacy"]},{"cell_type":"code","execution_count":15,"metadata":{},"outputs":[{"name":"stderr","output_type":"stream","text":["Traceback (most recent call last):\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 174, in _new_conn\n","    conn = connection.create_connection(\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 95, in create_connection\n","    raise err\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\connection.py\", line 85, in create_connection\n","    sock.connect(sa)\n","TimeoutError: [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 714, in urlopen\n","    httplib_response = self._make_request(\n","                       ^^^^^^^^^^^^^^^^^^^\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 403, in _make_request\n","    self._validate_conn(conn)\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 1053, in _validate_conn\n","    conn.connect()\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 363, in connect\n","    self.sock = conn = self._new_conn()\n","                       ^^^^^^^^^^^^^^^^\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connection.py\", line 179, in _new_conn\n","    raise ConnectTimeoutError(\n","urllib3.exceptions.ConnectTimeoutError: (<urllib3.connection.HTTPSConnection object at 0x00000200DA744210>, 'Connection to raw.githubusercontent.com timed out. (connect timeout=None)')\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 486, in send\n","    resp = conn.urlopen(\n","           ^^^^^^^^^^^^^\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\connectionpool.py\", line 798, in urlopen\n","    retries = retries.increment(\n","              ^^^^^^^^^^^^^^^^^^\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\urllib3\\util\\retry.py\", line 592, in increment\n","    raise MaxRetryError(_pool, url, error or ResponseError(cause))\n","urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='raw.githubusercontent.com', port=443): Max retries exceeded with url: /explosion/spacy-models/master/compatibility.json (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x00000200DA744210>, 'Connection to raw.githubusercontent.com timed out. (connect timeout=None)'))\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"<frozen runpy>\", line 198, in _run_module_as_main\n","  File \"<frozen runpy>\", line 88, in _run_code\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\spacy\\__main__.py\", line 4, in <module>\n","    setup_cli()\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\spacy\\cli\\_util.py\", line 92, in setup_cli\n","    command(prog_name=COMMAND)\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\click\\core.py\", line 1157, in __call__\n","    return self.main(*args, **kwargs)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\typer\\core.py\", line 778, in main\n","    return _main(\n","           ^^^^^^\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\typer\\core.py\", line 216, in _main\n","    rv = self.invoke(ctx)\n","         ^^^^^^^^^^^^^^^^\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\click\\core.py\", line 1688, in invoke\n","    return _process_result(sub_ctx.command.invoke(sub_ctx))\n","                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\click\\core.py\", line 1434, in invoke\n","    return ctx.invoke(self.callback, **ctx.params)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\click\\core.py\", line 783, in invoke\n","    return __callback(*args, **kwargs)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\typer\\main.py\", line 683, in wrapper\n","    return callback(**use_params)  # type: ignore\n","           ^^^^^^^^^^^^^^^^^^^^^^\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\spacy\\cli\\download.py\", line 36, in download_cli\n","    download(model, direct, sdist, *ctx.args)\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\spacy\\cli\\download.py\", line 70, in download\n","    compatibility = get_compatibility()\n","                    ^^^^^^^^^^^^^^^^^^^\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\spacy\\cli\\download.py\", line 94, in get_compatibility\n","    r = requests.get(about.__compatibility__)\n","        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\api.py\", line 73, in get\n","    return request(\"get\", url, params=params, **kwargs)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\api.py\", line 59, in request\n","    return session.request(method=method, url=url, **kwargs)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 589, in request\n","    resp = self.send(prep, **send_kwargs)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\sessions.py\", line 703, in send\n","    r = adapter.send(request, **kwargs)\n","        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"c:\\Users\\Ashish Yadav\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\requests\\adapters.py\", line 507, in send\n","    raise ConnectTimeout(e, request=request)\n","requests.exceptions.ConnectTimeout: HTTPSConnectionPool(host='raw.githubusercontent.com', port=443): Max retries exceeded with url: /explosion/spacy-models/master/compatibility.json (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x00000200DA744210>, 'Connection to raw.githubusercontent.com timed out. (connect timeout=None)'))\n"]}],"source":["!python -m spacy download en_core_web_lg"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":390},"executionInfo":{"elapsed":11844,"status":"error","timestamp":1692114616714,"user":{"displayName":"69. Ashish Yadav","userId":"08547316013865081816"},"user_tz":-330},"id":"0AeXBbNn428V","outputId":"6b8abc5d-a493-4b69-b983-5995a906d0ac"},"outputs":[],"source":["\n","# nlp = spacy.load(\"en_core_web_lg\")"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"SDI9rbZG49eI"},"outputs":[],"source":["# df['vector'] = df['Text'].apply(lambda x: nlp(x).vector)"]},{"cell_type":"code","execution_count":null,"metadata":{"executionInfo":{"elapsed":373,"status":"ok","timestamp":1692114691139,"user":{"displayName":"69. Ashish Yadav","userId":"08547316013865081816"},"user_tz":-330},"id":"Xp2mH21m5Mbt"},"outputs":[],"source":["# from sklearn.model_selection import train_test_split"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"MXpNROC95R6U"},"outputs":[],"source":["# x = df.vector\n","# y = df.labe_num\n","# # x_train,x_test,y_train,y_test = train_test_split(X,y,test_size=0.2,random_state=1)"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"BPYXsMW38OVS"},"outputs":[],"source":["# import numpy as np\n","# x_train_2d = np.stack(x_train) # array of array to 2d matrix of numpy\n","# x_test_2d = np.stack(x_test)"]},{"cell_type":"code","execution_count":null,"metadata":{"executionInfo":{"elapsed":5,"status":"ok","timestamp":1692115613615,"user":{"displayName":"69. Ashish Yadav","userId":"08547316013865081816"},"user_tz":-330},"id":"T8N_DlyF8rZa"},"outputs":[],"source":["# from sklearn.naive_bayes import MultinomialNB\n","# from sklearn.preprocessing import MinMaxScaler\n","# from sklearn.metrics import classification_report\n","# scaler = MinMaxScaler()\n","# scaled_x_train = scaler.fit_transform(x_train_2d)\n","# scaled_x_test = scaler.transform(x_test_2d)\n","# model = MultinomialNB()\n","# model.fit(x_train_2d,y_train)"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"GODAyoaA9ON2"},"outputs":[],"source":["# y_pred = model.predict(scaled_x_test)\n","# print(classification_report(y_test,y_pred))"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"h0CVVNBU9X2y"},"outputs":[],"source":["# from sklearn.neighbors import KNeighborsClassifier\n","# knn_model = KNeighborsClassifier(n_neighbors = 5,metric = \"euclidean\")\n","# knn_model.fit(x_train_2d,y_train)\n","# y_pred = model.predict(scaled_x_test)\n","# print(classification_report(y_test,y_pred))"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[{"name":"stderr","output_type":"stream","text":["Matplotlib is building the font cache; this may take a moment.\n"]}],"source":["from sklearn.metrics import confusion_matrix\n","import seaborn as sns\n","import matplotlib.pyplot as plt"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["cm = confusion_matrix(y_test,y_pred)"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["plt.figure(figsize = (12,6))\n","sns.heatmap(cm,annot=True)\n","plt.xlabel(\"Predicted\")\n","plt.ylabel(\"Actual\")\n","plt.title(\"Confusion Matrix\")\n","plt.show()"]}],"metadata":{"colab":{"authorship_tag":"ABX9TyMeo9kpJhF7s7A3clMiZ+Gv","provenance":[]},"kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.11.4"}},"nbformat":4,"nbformat_minor":0}
